tests/test.py.in

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python3
+
+import hashlib
+import os
+import subprocess
+import sys
+
+# turn on to regenerate all checksums
+REGENERATE_CHECKSUMS = False
+
+class NoInput(Exception):
+    pass
+
+class NoFileGenerated(Exception):
+    pass
+
+class UnexpectedFileGenerated(Exception):
+    pass
+
+class ChecksumMismatch(Exception):
+    pass
+
+# pass in plugin folder from cmake
+# built plugins should be in the same folder as the csound binary
+os.environ['OPCODE6DIR64'] = "$<TARGET_FILE_DIR:csound-bin>"
+os.environ['CS_USER_PLUGINDIR'] = "$<TARGET_FILE_DIR:csound-bin>"
+
+# directory: path where csd file is
+# filename: e.g. test, if file is test.csd
+# status: one of "passing", "indeterminate_output", or "no_output"
+#
+# - "no_output": generates no output
+# - "stalls": runs indefinitely or for too long
```

Comment:
Any test that stalls or never complete's is just a broken test. I don't think we should accept that as a valid test. Did you notice any tests doing that?

Reply:
You should comment them out in the cmake and leave a note to prevent it running rather than having a bail out option here imo.

Reply:
Ok, done!